src/ProjectSystem/Impl/IO/MsBuildFileSystemWatcher.cs

```diff
@@ -215,6 +215,9 @@ public sealed partial class MsBuildFileSystemWatcher : IDisposable {
 
             relativePath = PathHelper.MakeRelative(rootDirectory, fullPath);
             try {
+                if (!fileSystem.FileExists(fullPath)) {
```

Comment:
Race may occur right after this method returns `true`, and there is try/catch around this block already, so calling `FileExists` is useless. But we probably need to catch `UnauthorizedAccessException` in addition to `IOException`.